appengine/src/main/java/com/google/googleinterns/gscribe/GScribeConfiguration.java

```diff
@@ -16,8 +16,44 @@
 
 package com.google.googleinterns.gscribe;
 
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.googleinterns.gscribe.config.MySQLConfig;
 import io.dropwizard.Configuration;
 
+import javax.validation.Valid;
+import javax.validation.constraints.NotNull;
+
 public class GScribeConfiguration extends Configuration {
 
+    /**
+     * Reads the database configuration from the yaml file
+     * Needs - username, password, url, driver class
+     */
+    @Valid
+    @NotNull
+    private MySQLConfig mySQLConfig;
+
+    @Valid
+    @NotNull
+    private Environment environment;
+
+    @JsonProperty("database")
```

Comment:
Why is this required?

Reply:
This is used to read database credentials from the YAML file itself.